cmd/server/pactasrv/initiative_user_relationship.go

```diff
@@ -0,0 +1,75 @@
+package pactasrv
+
+import (
+	"context"
+
+	"github.com/RMI/pacta/cmd/server/pactasrv/conv"
+	"github.com/RMI/pacta/db"
+	"github.com/RMI/pacta/oapierr"
+	api "github.com/RMI/pacta/openapi/pacta"
+	"github.com/RMI/pacta/pacta"
+	"go.uber.org/zap"
+)
+
+// Returns all initiative user relationships for the user that the user has access to view
+// (GET /initiative/{id}/user-relationships)
+func (s *Server) ListInitiativeUserRelationshipsByUser(ctx context.Context, request api.ListInitiativeUserRelationshipsByUserRequestObject) (api.ListInitiativeUserRelationshipsByUserResponseObject, error) {
+	// TODO(#12) Implement Authorization
+	iurs, err := s.DB.InitiativeUserRelationshipsByUser(s.DB.NoTxn(ctx), pacta.UserID(request.UserId))
+	if err != nil {
+		return nil, oapierr.Internal("failed to retrieve initiative user relationships by user", zap.Error(err))
+	}
+	result, err := dereference(mapAll(iurs, conv.InitiativeUserRelationshipToOAPI))
+	if err != nil {
+		return nil, err
+	}
+	return api.ListInitiativeUserRelationshipsByUser200JSONResponse(result), nil
+}
+
+// Returns all initiative user relationships for the initiative that the user has access to view
+// (GET /initiative/user-relationships/{id})
+func (s *Server) ListInitiativeUserRelationshipsByInitiative(ctx context.Context, request api.ListInitiativeUserRelationshipsByInitiativeRequestObject) (api.ListInitiativeUserRelationshipsByInitiativeResponseObject, error) {
+	// TODO(#12) Implement Authorization
+	iurs, err := s.DB.InitiativeUserRelationshipsByInitiatives(s.DB.NoTxn(ctx), pacta.InitiativeID(request.InitiativeId))
+	if err != nil {
+		return nil, oapierr.Internal("failed to retrieve initiative user relationships by user", zap.Error(err))
+	}
+	result, err := dereference(mapAll(iurs, conv.InitiativeUserRelationshipToOAPI))
+	if err != nil {
+		return nil, err
+	}
+	return api.ListInitiativeUserRelationshipsByInitiative200JSONResponse(result), nil
+}
+
+// Returns the initiative user relationship from this id, if it exists
+// (GET /initiative/{initiativeId}/user-relationship/{userId})
```

Comment:
This API surface seems really broad to me, what's the motivation for having three different endpoints that return a user's relationship to an initiative? Unless we expect the list of relationships to be exceptionally large (>1,000), I'd just let the client filter for the stuff they're looking for.

More API surface means more stuff to secure, and more DB functions/tests/code to maintain overall.

Reply:
Yes, they will be large. There is a public initiative, which anyone will have the ability to join, and may be segmented by region (i.e. can't be handled as a standalone case).